src/mmrelay/main.py

```
@@ -209,9 +276,10 @@ def signal_handler() -> None:
         # On Windows, we can't use add_signal_handler, so we'll handle KeyboardInterrupt
         pass
 
-    # Start connection health monitoring using getMetadata() heartbeat
-    # This provides proactive connection detection for all interface types
-    _ = asyncio.create_task(meshtastic_utils.check_connection())
+    if active_backend_name == "meshtastic":
+        # Start connection health monitoring using getMetadata() heartbeat
+        # This provides proactive connection detection for all interface types
+        _ = asyncio.create_task(meshtastic_utils.check_connection())
```

Ruff (0.14.13)

Store a reference to the return value of `asyncio.create_task`

(RUF006)